client/asset/dcr/dcr.go

```diff
@@ -1882,67 +1885,98 @@ func (dcr *ExchangeWallet) findRedemptionsInMempool(contractOutpoints []outPoint
+// redemption finder(s) and these may re-call dcr.FindRedemption to restart find
+// redemption attempts for any of these contracts.
+func (dcr *ExchangeWallet) findRedemptionsInBlockRange(startBlockHeight, endBlockHeight int64, contractOutpoints []outPoint) {
+	totalContracts := len(contractOutpoints)
 	dcr.log.Debugf("finding redemptions for %d contracts in blocks %d - %d",
-		contractsCount, startBlock.height, endBlock.height)
+		totalContracts, startBlockHeight, endBlockHeight)
 
-	nextBlockHash := startBlock.hash
-	var lastScannedBlockHeight int64
+	var lastScannedBlockHeight int64 = -1 // we've not scanned any block yet
 	var totalFound, totalCanceled int
 
 rangeBlocks:
-	for nextBlockHash != nil && lastScannedBlockHeight < endBlock.height {
-		blk, err := dcr.node.GetBlockVerbose(dcr.ctx, nextBlockHash, true)
-		if err != nil {
-			// Redemption search for this set of contracts is compromised. Notify
-			// the redemption finder(s) of this fatal error and cancel redemption
-			// search for these contracts. The redemption finder(s) may re-call
-			// dcr.FindRedemption to restart find redemption attempts for any of
-			// these contracts.
-			err = fmt.Errorf("error fetching verbose block %s: %w", nextBlockHash, translateRPCCancelErr(err))
+	for blockHeight := startBlockHeight; blockHeight <= endBlockHeight; blockHeight++ {
+		// Get the hash for this block.
+		blockHash, err := dcr.node.GetBlockHash(dcr.ctx, blockHeight)
+		if err != nil { // unable to get block hash is a fatal error
+			err = fmt.Errorf("unable to get hash for block %d: %w", blockHeight, translateRPCCancelErr(err))
+			dcr.fatalFindRedemptionsError(err, contractOutpoints)
+			return
+		}
+
+		// Combine the p2sh scripts for all contracts (excluding contracts whose redemption
+		// have been found) to check against this block's v2cfilters.
+		dcr.findRedemptionMtx.RLock()
+		contractP2SHScripts := make([][]byte, 0)
+		for _, contractOutpoint := range contractOutpoints {
+			if req, stillInQueue := dcr.findRedemptionQueue[contractOutpoint]; stillInQueue {
+				contractP2SHScripts = append(contractP2SHScripts, req.contractP2SHScript)
+			}
+		}
+		dcr.findRedemptionMtx.RUnlock()
+
+		// Check if any of the above p2sh scripts is possibly included in this block.
+		hit, err := dcr.blockMaybeContainsScripts(blockHash.String(), contractP2SHScripts)
+		if err != nil { // error checking a block's cfilters is a fatal error
+			err = fmt.Errorf("error checking v2cfilters for block %d (%s) for likely contract inclusion: %w",
+				blockHeight, blockHash, translateRPCCancelErr(err))
+			dcr.fatalFindRedemptionsError(err, contractOutpoints)
+			return
+		}
+		if !hit {
+			lastScannedBlockHeight = blockHeight
+			continue // block does not reference any of these contracts, continue to next block
+		}
+
+		// Pull the block info to confirm if any of its inputs spends a contract of interest.
+		blk, err := dcr.node.GetBlockVerbose(dcr.ctx, blockHash, true)
```

Comment:
This is fine, but just a note: we don't really need getblock with either verbose=true or verboseTx=true since with a block's bytes we could deserialize and work on the `MsgTx`s.  No changes requests though.  It's going to be a headache to change, esp. on account of the helper methods needing `TxRawResult`s.  Besides this is not a hot path in dex client code.

Reply:
Will add a todo to that effect.